rust/agama-lib/src/scripts/store.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) [2024] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use crate::{base_http_client::BaseHTTPClient, error::ServiceError};
+
+use super::{client::ScriptsClient, settings::ScriptsConfig, Script, ScriptConfig, ScriptsGroup};
+
+pub struct ScriptsStore {
+    client: ScriptsClient,
+}
+
+impl ScriptsStore {
+    pub fn new(client: BaseHTTPClient) -> Self {
+        Self {
+            client: ScriptsClient::new(client),
+        }
+    }
+
+    pub fn new_with_client(client: ScriptsClient) -> Result<Self, ServiceError> {
+        Ok(Self { client })
```

Comment:
how it can fail? this looks kind of strange.

Reply:
Yes, you are right. It behaves in this way for consistency with other clients. I already told to @mchf on IRC that we should change this behavior for all the clients but as a follow-up PR.

Reply:
Actually, it is not even needed 😅. We need to check the other stores too.

Reply:
Actually, in this case it is not even needed at all (perhaps when we extend the tests suite).